### PR TITLE
MM-18993 Migrate 'components/toggle.jsx' to TypeScript

### PR DESCRIPTION
**Pull request by anchepiece**
#### Summary

This pull request migrates components/toggle.jsx to TypeScript.
All tests pass locally, suggestions on improvement welcome.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12436

### CODE REVIEW

File: components/toggle.tsx [hunk 1]
```diff
@@ -2,33 +2,36 @@
...
-export default function Toggle({onToggle, toggled, onText, offText, disabled}) {
+type Props = {
+    onToggle: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    toggled: boolean,
+    disabled?: boolean,
+    onText: React.ReactNode,
+    offText: React.ReactNode,
+}
+
+export default function Toggle(props: Props) {
```
Comment (jespino) [inline components/toggle.tsx]: You can be more explicit here about the type of the function using the `React.FC` type, see an example here: https://github.com/mattermost/mattermost-webapp/blob/master/components/widgets/badges/guest_badge.tsx#L14

File: components/toggle.tsx [hunk 2]
```diff
@@ -2,33 +2,36 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
-export default function Toggle({onToggle, toggled, onText, offText, disabled}) {
+type Props = {
+    onToggle: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    toggled: boolean,
```
Comment (jespino) [inline components/toggle.tsx]: `toggled`, `onText` and `offText` were optional before, why now are mandatory?

File: components/toggle.tsx [hunk 3]
```diff
@@ -2,33 +2,36 @@
...
+
+export default function Toggle(props: Props) {
+    const {onToggle, toggled, disabled, onText, offText} = props;
     return (
         <button
             type='button'
             onClick={onToggle}
             className={`btn btn-lg btn-toggle ${toggled && 'active'} ${disabled && 'disabled'}`}
             aria-pressed={toggled ? 'true' : 'false'}
-            autoComplete='off'
```
Comment (jespino) [inline components/toggle.tsx]: I like it

File: components/toggle.tsx [hunk 4]
```diff
@@ -2,33 +2,36 @@
...
         >
             <div className='handle'/>
             {text(toggled, onText, offText)}
         </button>);
 }
 
-function text(toggled, onText, offText) {
+function text(
```
Comment (jespino) [inline components/toggle.tsx]: You can maintain everything in the same line here, probably is more consistent with the rest of our code base.

File: components/toggle.tsx (hunk 2 above)
Comment (anchepiece) [reply components/toggle.tsx]: Fixed

**Comment by jespino** [comment]
@anchepiece you have to fix some lint errors, but the rest of the code looks good :)

**Comment by anchepiece** [comment]
@jespino Linting issues addressed in the last commit. Thanks!

**Comment by jespino** [comment]
@anchepiece merged!, thanks! :tada:

**Pull request merged**